R/DEoptim_fns.R

```diff
@@ -150,61 +151,79 @@ runDEoptim <- function(landscape,
     ## Make cluster with just one worker per machine --> don't need to do these steps
     #     multiple times per machine, if not all 'localhost'
     revtunnel <- FALSE
+    neededPkgs <- c("kSamples", "magrittr", "raster", "data.table",
+                    "SpaDES.tools", "fireSenseUtils")
+
+    # browser()
     if (!identical("localhost", unique(cores))) {
-      revtunnel <- ifelse(all(cores == "localhost"), FALSE, TRUE)
+      repos <- c("https://predictiveecology.r-universe.dev", getOption("repos"))
 
-      coresUnique <- setdiff(unique(cores), "localhost")
-      message(
-        "Making sure packages with sufficient versions installed and loaded on: ",
-        paste(coresUnique, collapse = ", ")
-      )
-      st <- system.time({
-        cl <- parallelly::makeClusterPSOCK(coresUnique, revtunnel = revtunnel, rscript_libs = libPath)
-      })
-      packageVersionFSU <- packageVersion("fireSenseUtils")
-      packageVersionST <- packageVersion("SpaDES.tools")
-      clusterExport(cl, list("libPath", "logPath", "packageVersionFSU", "packageVersionST"), envir = environment())
-
-      parallel::clusterEvalQ(
-        cl,
-        {
-          # If this is first time that packages need to be installed for this user on this machine
-          #   there won't be a folder present that is writable
-          if (!dir.exists(libPath)) {
-            dir.create(libPath, recursive = TRUE)
+      aa <- Require::pkgDep(unique(c("dqrng", "PredictiveEcology/SpaDES.tools@development",
```

Comment:
need to be careful allowing Require to install update anything without specifying exact versions (i.e., those installed on the main node); otherwise we get mismatched packages on each node.

Can you confirm that the versions installed are exactly the same, and that it's not just grabbing the latest versions of everything?

Reply:
It isn't installing them ... this is just determining which to use. It copies them from the source folder. Previously this was not checking versions. Now it is using the exact version because they are being copied.